packages/cloudflare/src/index.ts

```diff
@@ -88,6 +90,50 @@ export default function createIntegration(args?: Options): AstroIntegration {
 								disabled: !args?.wasmModuleImports,
 							}),
 							chunkAnalyzer.getPlugin(),
+							{
+								name: 'prerender-scanner',
+								enforce: 'post',
+								generateBundle(_, bundle) {
+									for (const chunk of Object.values(bundle)) {
+										if (chunk.type !== 'chunk') continue;
+										if (chunk.dynamicImports.some((entry) => entry.includes('prerender'))) {
+											prerenderImports.push(chunk.fileName);
+										}
+									}
+								},
+							},
+							{
+								name: 'prerender-worker',
+								enforce: 'post',
+								generateBundle(_, bundle) {
+									for (const chunk of Object.values(bundle)) {
+										if (chunk.type !== 'chunk') continue;
+										if (chunk.name === '_@astrojs-ssr-virtual-entry') {
+											chunk.dynamicImports = chunk.dynamicImports.filter(
+												(entry) => !prerenderImports.includes(entry)
+											);
+											for (const page of prerenderImports) {
+												const importRegex = new RegExp(
+													`^const (_page\\d) = \\(\\) => import\\('.\\/${page}'\\);$\\n`,
+													'gm'
+												);
+												let pageId: string | undefined;
+												const matches = chunk.code.matchAll(importRegex);
+												for (const match of matches) {
+													if (match[1]) {
+														pageId = match[1];
+													}
+												}
```

Comment:
It seems this will find the last occurrence of the import. What if the same chunk is imported more than once? Is there a guarantee that it won't happen (I know we're working with very few guarantees here)

Reply:
Initially I thought I needed two plugins because I first need to check all chunks in the bundle before I can mutate the entry. And the order was not right, but just looping twice inside the same plugin would work!